vars/capsulePipelineWrapper.groovy

```diff
@@ -1,245 +1,485 @@
-                        environment {
-                            FLAGS = "--out-dir '${env.WORKSPACE}/networks-internal/stagnet3/live-config'"
-                        }
-                        steps {
-                            script {
-                                writeConfigs('live config (git / networks-internal)')
-                            }
-                        }
-                    }
-                    stage('Template s3 config') {
-                        environment {
-                            FLAGS = '--update-network'
-                        }
-                        steps {
-                            script {
-                                writeConfigs('s3')
-                            }
-                        }
-                    }
-                    stage('Write to git') {
-                        steps {
-                            makeCommit(
-                                makeCheckout: false,
-                                directory: 'networks-internal',
-                                branchName: 'live-config-update',
-                                commitMessage: '[Automated] live config update',
-                            )
-                        }
-                    }
-                }
+
+            steps {
+              gitClone([
+                  credentialsId: env.GITHUB_SSH_CREDS,
+                  url: 'git@github.com:vegaprotocol/vegacapsule.git',
+                  branch: params.VEGACAPSULE_VERSION,
+                  directory: 'vegacapsule'
+              ])
+
+              dir('vegacapsule') {
+                sh 'go build -o vegacapsule .'
+                sh 'mv vegacapsule ../bin'
+              }
+              sh 'chmod +x bin/vegacapsule'
+              sh 'vegacapsule --help'
             }
-            stage('Restart Network') {
-                when {
-                    expression {
-                        (params.ACTION == 'RESTART' || params.ACTION == 'START') && params.UNSAFE_RESET_ALL
-                    }
-                }
-                steps {
-                    sh "vegacapsule nodes unsafe-reset-all --remote --home-path '${env.CONFIG_HOME}'"
+          }
+
+          stage('Download capsule') {
+            when {
+              not {
+                expression {
+                  params.BUILD_CAPSULE
                 }
+              }
             }
-            stage('Start Network') {
-                when {
-                    expression {
-                        params.ACTION == 'START' || params.ACTION == 'RESTART'
-                    }
-                }
-                steps {
-                    sh "vegacapsule network start --home-path '${env.CONFIG_HOME}'"
-                }
+            steps {
+              withGHCLI('credentialsId': env.GITHUB_CREDS) {
+                sh 'gh release --repo vegaprotocol/vegacapsule download ' + params.VEGACAPSULE_VERSION + ' --pattern "*linux*"'
```

Comment:
You did a function for Vega binary using GH release command, let's parametrize it and call it here too

Reply:
fixed